darasimiolaifa/kabu-kabu-microservice
# chore(api-gateway): Setup api gateway

## What does this PR do?
This PR sets up the api gateway for the kabu-kabu microservices app

## What major activities were carried out?
- download the fast-gateway module
- set up travis, codeclimate and coveralls to watch the repo
- set up eslint and babel to ensure and enforce standards
- create stories for the setup
- create the gateway entry file and configure it
- write test to test the gateway using an index route

## How can this be manually tested?
- Clone repo by running `git clone https://github.com/darasimiolaifa/kabu-kabu-microservice.git`
- Run the command `cd kabu-kabu-microservice/api-gateway`
- Run `npm install`
- Run `npm start`
- In Postman, send a `GET` request to this url `http://localhost:8080`, you should get the following response:
```json
  {
    "status": 200,
    "message": "Welcome to the kabu-kabu app."
  }
```

## Any background information to provide?
- It is important to cd into the folder that holds the app for the gateway code, as this is built with the microservices architecture where each service is a standalone app. Else the npm scripts will throw an error.

PT story link: [#169062047](https://www.pivotaltracker.com/story/show/169062047)
